ank/src/cli_commands/server_connection.rs

```diff
@@ -0,0 +1,800 @@
+            } else {
+                // [impl->swdd~cli-stores-unexpected-message~1]
+                self.missed_from_server_messages.push(server_message);
+            };
+        }
+    }
+
+    pub fn take_missed_from_server_messages(&mut self) -> Vec<FromServer> {
+        take(&mut self.missed_from_server_messages)
+    }
+}
+
+#[derive(Debug)]
+pub enum ServerConnectionError {
+    ExecutionError(String),
+}
+
+//////////////////////////////////////////////////////////////////////////////
+//                 ########  #######    #########  #########                //
+//                    ##     ##        ##             ##                    //
+//                    ##     #####     #########      ##                    //
+//                    ##     ##                ##     ##                    //
+//                    ##     #######   #########      ##                    //
+//////////////////////////////////////////////////////////////////////////////
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use common::{
+        commands::{
+            CompleteStateRequest, Error, RequestContent, Response, ResponseContent,
+            UpdateStateRequest, UpdateStateSuccess, UpdateWorkloadState,
+        },
+        from_server_interface::FromServer,
+        objects::{
+            CompleteState, ExecutionState, ExecutionStateEnum, RunningSubstate, State,
+            StoredWorkloadSpec, WorkloadInstanceName, WorkloadState,
+        },
+        to_server_interface::ToServer,
+    };
+    use tokio::sync::mpsc::Receiver;
+
+    use super::ServerConnection;
+
+    const WORKLOAD_NAME_1: &str = "workload_1";
+    const WORKLOAD_NAME_2: &str = "workload_2";
+    const AGENT_A: &str = "agent_A";
+    const RUNTIME: &str = "runtime";
+    const REQUEST: &str = "complete_state_request";
+    const FIELD_MASK: &str = "field_mask";
+    const ID: &str = "id";
+
+    fn complete_state_1() -> CompleteState {
+        CompleteState {
+            desired_state: State {
+                workloads: [workload(WORKLOAD_NAME_1)].into(),
+                ..Default::default()
+            },
+            ..Default::default()
+        }
+    }
+
+    fn complete_state_2() -> CompleteState {
+        CompleteState {
+            desired_state: State {
+                workloads: [workload(WORKLOAD_NAME_2)].into(),
+                ..Default::default()
+            },
+            ..Default::default()
+        }
+    }
+
+    fn workload(workload_name: &str) -> (String, StoredWorkloadSpec) {
+        (
+            workload_name.into(),
+            StoredWorkloadSpec {
+                agent: AGENT_A.into(),
+                runtime: RUNTIME.into(),
+                ..Default::default()
+            },
+        )
+    }
+
+    fn instance_name(workload_name: &str) -> WorkloadInstanceName {
+        format!("{workload_name}.{ID}.{AGENT_A}")
+            .try_into()
+            .unwrap()
+    }
```

Comment:
I do not understand want you exactly mean. Do you want to replace the functions above with existing `generate_test_...` functions or do you want to use the `generate_test_...` functions in the functions above?

I would not replace the functions above. All existing `generate_test_...` functions take multiple arguments, which would lead to a lot of repeated code in the test.

Using the `generate_test_...` functions in the functions above, would not simplify the code, as the generating of objects using `..Default::default()` is simple and short. I also think instantiating the objects directly is cleaner, as the `generate_test_...` functions take multiple arguments without clue, which argument is for which field. Using object instantiation, everything is named.

Reply:
I have another solution. I have replaced these functions:

```rust
fn complete_state_1() -> CompleteState {
        CompleteState {
            desired_state: State {
                workloads: [workload(WORKLOAD_NAME_1)].into(),
                ..Default::default()
            },
            ..Default::default()
        }
    }

    fn complete_state_2() -> CompleteState {
        CompleteState {
            desired_state: State {
                workloads: [workload(WORKLOAD_NAME_2)].into(),
                ..Default::default()
            },
            ..Default::default()
        }
    }

    fn workload(workload_name: &str) -> (String, StoredWorkloadSpec) {
        (
            workload_name.into(),
            StoredWorkloadSpec {
                agent: AGENT_A.into(),
                runtime: RUNTIME.into(),
                ..Default::default()
            },
        )
    }
```

With one function:

```rust
    fn complete_state(workload_name: &str) -> CompleteState {
        CompleteState {
            desired_state: State {
                workloads: [(
                    workload_name.into(),
                    StoredWorkloadSpec {
                        agent: AGENT_A.into(),
                        runtime: RUNTIME.into(),
                        ..Default::default()
                    },
                )]
                .into(),
                ..Default::default()
            },
            ..Default::default()
        }
    }
```

What do you say to this solution?

Reply:
I have implemented to solution.